File: python/bubble_sort.py
sequence = [4, 3, 5, 0, 1]


def bubble_sort(sequence):
    swaps = 0
    for i in range(len(sequence)):
        for j in range(len(sequence)-1):
            if sequence[j] > sequence[j+1]:
                current = sequence[j]
                sequence[j] = sequence[j+1]
                sequence[j+1] = current
            print(sequence)
            swaps+=1
    result = sequence
    print(f"Final result: {result}")
    print(f"Swaps: {swaps}")



bubble_sort(sequence)




# def bubble_sort(sequence):
#     swaps = 0
#     for i in range(len(sequence)):
#         for j in range(len(sequence)-1):
#             print(sequence)
#             print(f'Comparing {sequence[i]} to {sequence[j]}')
#             if sequence[i] < sequence[j]:
#                 temp = sequence[i]
#                 sequence[i] = sequence[j]
#                 sequence[j] = temp
#                 swaps+=1
#     result = sequence
#     print(f"Final result: {result}")
#     print(f"Swaps: {swaps}")


# bubble_sort(sequence)

File: js/bubbleSort.js
var sequence = [4, 3, 5, 0, 1]
var swaps = 0

function bubbleSort(arr){
    let swapCount = 0
    for (let i = 0; i < arr.length;i++) {
        for (let j = 0; j < arr.length-1; j++){
            if (arr[j] > arr[j+1]){
                let current = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = current
                console.log(arr)
            }
            swapCount+=1;
        }
    }
    // let result = arr
    console.log(arr)
}


bubbleSort(sequence)
// console.log("Final result: " + result)
// console.log("Swaps: " + swaps)
